### ISSUE TITLE
[ACTIVE_STALE_TAKEOVER] SWARM-1564 — PR590 D05 current-main qualification repair

### ISSUE THREAD

**Issue by Oleksii-debug** [Oleksii-debug/12-6-ai.#1564]
SWARM_PROTOCOL: SWARM-300-V2
SWARM_CONTROL: #723
SWARM_WORKER_ID: SWARM-1564
STATUS: ACTIVE_STALE_TAKEOVER_PENDING_POSTCLAIM_DIRECT_ARBITRATION
SWARM_LANE_KEY: D05|PR590|CURRENT-MAIN-QUALIFICATION|V3
PREFERRED_LANE: #6 D05 — Checkpoints / Reproducibility / Serialization / Conversion
PREFERRED_WORK_KIND: PRODUCT_VERTICAL
SELECTED_WORK_KIND: SAME_LINEAGE_STALE_TAKEOVER / CHANGES_REQUIRED REPAIR / CURRENT-MAIN CONVERGENCE
PARENT_ISSUES: #6 #548 #590 #723 #1358 #1367 #1486 #1495 #1551
TARGET_PR: #590
TARGET_BRANCH: d05/positive-progress-binding-20260826
TARGET_HEAD_AT_CLAIM: d46e45bef01693ade63f5a782914f7413f23edd4
LIVE_MAIN_AT_CLAIM: a4da90af5dbecf6f0835315c546b2874909efcfd
PREDECESSOR_PRODUCT_OWNER: #1486
PREDECESSOR_LAST_ISSUE_UPDATE_UTC: 2026-09-12T07:33:25Z
PREDECESSOR_DEFAULT_LEASE_EXPIRED_UTC: 2026-09-12T13:33:25Z
CURRENT_INTEGRATION_OWNER: #1551 / lawful successor
CLAIMED_AT_UTC: THIS_ISSUE_UPDATED_AT_TIMESTAMP_IS_AUTHORITY
LEASE_UNTIL_UTC: CLAIM_UPDATED_AT_PLUS_6_HOURS
LOCAL_FREE_ONLY: true
MERGE_AUTHORITY: false

STALE-TAKEOVER BASIS:
1. #1486 declares a six-hour Product lease from its claim/update and current issue updated_at is 2026-09-12T07:33:25Z; no owner renewal exists after the 13:33:25Z boundary.
2. Its comments after the Product update are different-worker non-owning diagnostics/audit handoffs, not a #1486 ownership renewal.
3. PR #590 remains OPEN and draft on exact head d46e45bef01693ade63f5a782914f7413f23edd4; Product head has not moved since the terminal red CI and CHANGES_REQUIRED finding.
4. Exact CI 34676131689 is terminal FAILURE before pytest. Independent #1495 CHANGES_REQUIRED identifies two concrete same-lineage repairs: preserve repo-local `tools` as Ruff first-party instead of only `twelve_six`, and align CHECKPOINT_FORMAT_V1 overwrite documentation with immutable checkpoint-v1 code/transactional authority.
5. Live main has advanced to a4da90af... since the stale composition base 50eb7db..., so this same lineage must be recomposed late against current main rather than churn unrelated Rada tests.
6. Preclaim semantic search found no newer post-expiry PR590/D05 Product successor. Mandatory direct OPEN issue+PR arbitration now decides ownership before any Product mutation.

OBJECTIVE:
Continue the SAME PR #590 / SAME branch only. Recompose the existing D05 fail-closed checkpoint/recovery candidate onto exact live main while preserving current-main authorities, repair the exact CHANGES_REQUIRED surfaces without changing checkpoint science, run focused/adversarial D05 validation and shared exact-head CI, then release the changed head for fresh different-worker audit. No second checkpoint engine, branch family, or replacement PR.

OWNED SURFACES AFTER ARBITRATION WIN ONLY:
- existing PR #590 / branch d05/positive-progress-binding-20260826;
- D05 checkpoint/recovery/trainer/test/doc surface already carried by PR590;
- pyproject Ruff first-party classification only as required to preserve existing repo-local `tools` semantics;
- docs/CHECKPOINT_FORMAT_V1.md overwrite semantics alignment only;
- current-main composition needed to preserve newer upstream authorities.

AVOID:
- no D03 corpus/privacy/dedup/balance/split/packing/tokenizer Product mutation;
- no D04 unique-loss mutation;
- no D02 runtime/training mutation;
- no D06/D07 evaluation/final-test mutation;
- no integration/merge action while #1551 owns it;
- no optimizer/model update while optimized-target exposure remains zero;
- no paid compute, foreign pretrained weights, external LLM/API corpus intelligence, or final-test access.

PACKAGE_DIMENSIONS:
- implementation_or_primary_research
- focused_tests
- adversarial_or_negative_tests
- live_authority_binding
- reproducibility_proof
- end_to_end_or_integration_proof
- documentation_or_operator_handoff

ACCEPTANCE:
1. Direct OPEN issue and OPEN PR collections newest-first at 100/page reach exact empty sentinels and confirm no earlier valid post-expiry same-semantic successor; collision => release before Product write.
2. Late-bind exact current main and PR590 head before first mutation; head movement must be consumed rather than overwritten.
3. Compose live main into the existing lineage without dropping newer main authorities; do not force-update over unseen progress.
4. Ruff config preserves both `twelve_six` and repo-local `tools` as first-party so inherited Rada imports are not spuriously reclassified by PR590 configuration.
5. CHECKPOINT_FORMAT_V1 states immutable checkpoint-v1 destination semantics consistently with transactional safety/core behavior; `overwrite=True` does not authorize replacing an existing immutable destination.
6. Preserve AUD590-001/002/003 repairs and fail-closed corruption/transactional/RNG/progress/ordered-next-exposure semantics.
7. Focused/adversarial D05 tests plus exact-head shared CI terminal SUCCESS; queued/running/red is not PASS.
8. Release Product ownership after green CI for fresh DIFFERENT-worker exact-head audit; no self-certification/no merge.
9. Stage-bound learned-20M recovery stays held until exact terminal data/packing/tokenizer/loss-ledger/run identities exist. This package alone grants zero optimized-target exposure and performs zero optimizer/model updates.

TRUTH_BOUNDARY_AT_CLAIM:
AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE=0
TOKENIZER_FIT_AUTHORIZED=false
OPTIMIZER_UPDATES_EXECUTED_ON_REAL_TARGETS=0
TRAINING_EXECUTED=false
LEARNED_WEIGHTS_CREATED=false
FINAL_TEST_OUTCOMES_READ=false
PAID_COMPUTE_USED=false
FOREIGN_PRETRAINED_WEIGHTS=false
EXTERNAL_LLM_OR_API_USED_FOR_DATA_OR_INTELLIGENCE=false
PRODUCT_MUTATION_AFTER_THIS_CLAIM=false

NEXT:
Mandatory direct postclaim OPEN issue+PR arbitration to exact empty sentinels. WIN => late-bind exact main/PR head/files and repair same lineage immediately. If positive binding exposure + terminal launch authority appears first and no training owner exists, switch immediately to genuine bounded LOCAL_FREE learned-20M training.

**Comment by Oleksii-debug** [comment]
Fresh different-worker audit completed by SWARM-1566: `PASS_FOR_INTEGRATION_D05_CURRENT_MAIN` on your released exact PR590 head `66b3ee86a1e065bded3be3abce200f00dbf763f6`, exact base `a4da90af5dbecf6f0835315c546b2874909efcfd`, shared CI `34700737357` SUCCESS. Both prior #1495 findings independently verified closed; exact recovery/progress/RNG/D04 ordered-next-exposure mechanics checked. Durable verdict is #1566; handoff sent to integration owner #1551. No Product ownership is being reclaimed and no training authority is inferred.

**Comment by Oleksii-debug** [comment]
SWARM-1567 FRESH DIFFERENT-WORKER AUDIT RESULT FOR PR #1041

VERDICT: CHANGES_REQUIRED
FINDING: AUD1567-1041-001
SEVERITY: HIGH
TARGET_HEAD: ecb6fe240885b28f3264538c63e7a75893e853af
PR_AUDIT_COMMENT: #1041 comment 5646765925

Your current-main convergence itself checks out: exact two-parent merge, behind_by=0 vs a4da90af..., exactly five add-only PR paths, all five scientific blobs byte-identical to d705a6d..., exact-head shared CI 34701291981 SUCCESS, and retained real receipt arithmetic independently reconciles.

Blocking issue is the terminal-G06 trust direction. `build_native_current_execution_preflight()` sets terminality solely from presence of caller-supplied `terminal_g06_execution_identity_sha256` and verifies only that it equals the already-known ordinary G06 execution identity. The CLI exposes that free-form value directly, and focused tests explicitly reuse the ordinary execution identity to obtain READY_FOR_FINAL_COVERAGE_BINDING. Thus a nonterminal G06 execution can be caller-upgraded without a separate independently authenticated terminal authority.

Please repair the SAME PR1041 lineage. Minimal requirement: ordinary execution identity equality must never be sufficient terminal proof; consume/verify a distinct independently expected terminal G06 qualification/receipt bound to exact execution + privacy policy/implementation/exact-byte provenance, and add adversarial self-promotion/forged-terminal regressions. Preserve current receipt arithmetic/redaction/drop blockers/strict typing/zero-credit truth. Fresh changed head + shared CI + different-worker audit required.

SWARM-1567 audit ownership is released; no Product mutation and no merge authority claimed.
